pkg/kubelet/cpumanager/state/state_mem.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/kubelet/cpuset"
+)
+
+type stateMemory struct {
+	assignments   map[string]cpuset.CPUSet
+	defaultCPUSet cpuset.CPUSet
+	pressure      bool
+}
+
+var _ State = &stateMemory{}
+
+// NewMemoryState creates new State for keeping track of cpu/pod assignment
+func NewMemoryState() State {
```

Comment:
On second look, we do need synchronization here. We already have concurrency hazard where the reconcile loop is reading `stateMemory.assignments` while the policy may be writing to the same map.